mlx/backend/metal/allocator.cpp

```diff
@@ -152,6 +153,8 @@ MetalAllocator::MetalAllocator()
       gc_limit_(0.95 * device_->recommendedMaxWorkingSetSize()) {}
 
 Buffer MetalAllocator::malloc(size_t size, bool allow_swap /* = false */) {
+  auto thread_pool = metal::new_scoped_memory_pool();
```

Comment:
Does it make sense to create this when we initialize the allocator or does that mess up the order of the operations?

Reply:
You don't want it scoped to the allocator -- an autoreleasepool is sort of like a scope in C++.  You want things to be collected/deallocated when the scope exits.  If it is in the allocator then it will live as long as the allocator.

Order of operations-wise it would be fine as long as the allocator was not created during static init (and I suspect it is).

Reply:
It is and thanks, that makes sense.

Reply:
It seems to help the benchmark if I move this to L180 which is the only place that we interact with Metal in this function.

We shouldn't need a pool if we are getting something from the cache (?), but could you check that makes sense?

Reply:
Makes sense -- I think it might need to be on L172 because it is likely needed if we need to release some of the memory.

Reply:
Moved

Reply:
Good point!